homeassistant/components/soma/cover.py

```diff
@@ -21,39 +44,173 @@ async def async_setup_entry(
     """Set up the Soma cover platform."""
 
     devices = hass.data[DOMAIN][DEVICES]
+    entities = []
+
+    for device in devices:
+        # Assume a shade device if the type is not present in the api response (Connect <2.2.6)
+        if "type" in device and device["type"].lower() == "tilt":
+            entities.append(SomaTilt(device, hass.data[DOMAIN][API]))
+        else:
+            entities.append(SomaShade(device, hass.data[DOMAIN][API]))
+
+    async_add_entities(entities, True)
 
-    async_add_entities(
-        [SomaCover(cover, hass.data[DOMAIN][API]) for cover in devices], True
-    )
 
+class SomaTilt(SomaEntity, CoverEntity):
+    """Representation of a Soma Tilt device."""
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_BLIND
 
-class SomaCover(SomaEntity, CoverEntity):
-    """Representation of a Soma cover device."""
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return (
+            SUPPORT_OPEN_TILT
+            | SUPPORT_CLOSE_TILT
+            | SUPPORT_STOP_TILT
+            | SUPPORT_SET_TILT_POSITION
+        )
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return the current cover tilt position."""
+        return self.current_position
+
+    @property
+    def is_closed(self):
+        """Return if the cover tilt is closed."""
+        return self.current_position == 0
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], 100)
+        if is_api_response_success(response):
+            self.set_position(0)
+        else:
+            raise HomeAssistantError(
+                MSG_DEVICE_UNREACHABLE % (self.name, response["msg"])
+            )
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], -100)
+        if is_api_response_success(response):
+            self.set_position(100)
+        else:
+            raise HomeAssistantError(
+                MSG_DEVICE_UNREACHABLE % (self.name, response["msg"])
```

Comment:
Please use `f-strings` here

Reply:
Maybe make it a function so you can keep it DRY

Reply:
Fixed

Reply:
Also I have changed the error messages to be different for each action. Therefore I didn't make a function.